tests/unit/document_tests.py

```diff
@@ -137,6 +139,22 @@ def test_document_exists(self):
         doc.create()
         self.assertTrue(doc.exists())
 
+    def test_document_exists_raises_httperror(self):
+        """
+        Test document exists raises an HTTPError.
+        """
+        # Mock HTTPError when running against CouchDB and Cloudant
+        resp = requests.Response()
+        resp.status_code = 400
+        self.client.r_session.head = mock.Mock(return_value=resp)
+        with self.assertRaises(requests.HTTPError) as cm:
+            doc = Document(self.db)
+            doc['_id'] = 'julia006'
```

Comment:
Again, I would move these 2 lines above the `with...` for the same reason mentioned above.